app/views/unattended/provisioning_templates/snippet/disk_enc_clevis_tang.erb

```diff
@@ -0,0 +1,87 @@
+<%#
+kind: snippet
+name: disk_enc_clevis_tang
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Binds encrypted root directory ('/') utilizing Clevis to Tang server(s) for
+  decryption. The first parent device containing a LUKS container will be used.
+  The temporary passphrase will be removed afterwards. Currently, only Red Hat
+  family and Ubuntu operating systems are supported.
+-%>
+<%
+  passphrase = host_param('disk_enc_passphrase') || 'linux'
+  tang_server_list = []
+  packages_redhat = "clevis clevis-luks clevis-systemd clevis-dracut"
+  packages_ubuntu = "clevis clevis-luks clevis-systemd clevis-initramfs"
+
+  unless host_param('disk_enc_tang_servers').blank?
+    if host_param('disk_enc_tang_servers').is_a?(String)
+      tang_server_list = [host_param('disk_enc_tang_servers')]
+    else
+      tang_server_list = host_param('disk_enc_tang_servers')
+    end
+  end
+-%>
+
+<% if (@host.operatingsystem.family == 'Redhat' or @host.operatingsystem.name == 'Ubuntu') and unless tang_server_list.blank? %>
+
+cat > /tmp/rootdir-luks-device.sh << "EOF"
+#!/bin/sh
+#
+# Author Jan Löser <loeser@atix.de>
+# Published under the GNU Public Licence 3
+#
+# This scripts tries to find the 1st LUKS device for / (root directory).
+#
+set -o pipefail
+
+rootdev=$(df / --output=source | tail -n1)
+targetdev=$(readlink -f $rootdev)
+slavedev=$targetdev
+
+while : ; do
+    /sbin/cryptsetup luksDump $slavedev &>/dev/null && echo $slavedev && exit 0
+    set -e
+    slave=$(find /sys/class/block/$(basename $slavedev)/slaves -type l | head -n1)
+    slavedev=$(find /dev -name "$(basename $slave)" | head -n1)
+    set +e
+done
+
+exit 1
+EOF
+
+# needs bash here because Ubuntu's sh (dash) doesn't support `-o pipefail` option
+luksdev=$(bash /tmp/rootdir-luks-device.sh)
+
+if [[ -n "$luksdev" ]]; then
+  echo "LUKS device found for '/': $luksdev"
+
+<% if @host.operatingsystem.family == 'Redhat' %>
+  if [ -f /usr/bin/dnf ]; then
+    dnf -y install <%= packages_redhat %>
+  else
+    yum -t -y install <%= packages_redhat %>
+  fi
+<% elsif @host.operatingsystem.name == 'Ubuntu' %>
+  apt -y install clevis clevis-luks clevis-initramfs clevis-systemd
+<% end %>
+
+<% for tang_server in tang_server_list %>
+  echo '<%= passphrase %>' | clevis luks bind -y -k - -d $luksdev tang '{"url": "<%= tang_server %>"}'
+<% end %>
+  echo '<%= passphrase %>' | cryptsetup luksRemoveKey $luksdev
+  systemctl enable clevis-luks-askpass.path
+  systemctl enable remote-cryptsetup.target
+
+<% if @host.operatingsystem.family == 'Redhat' %>
```

Comment:
Is something else necessary for other OS, too?

Reply:
I don't understand the question.

Reply:
There are two special cases implemented for Redhat and Ubuntu. I wonder, if something else need to be done for Alma, Rocky, Debian, etc?

Reply:
At the moment only the special cases are tested. There might be more operating systems that could use this snippet (with updated OS conditions), but this was not tested yet. For SUSE for example, this snippet would also work (technically) but the test system wasn't able to decrypt and boot up successfully. Debugging these issues on all operating systems is quite expensive and can be done in future.

Reply:
If this was not tested for other OS, then we should maybe just specifiy the related OS in 
https://github.com/theforeman/foreman/pull/9887/files#diff-74ce7a8de62f4fd3cd241aed74d88610178ae49c381d7b7cc8b3deb2908c3406R6